service/storage_service.cc

```diff
@@ -3849,8 +3849,31 @@ future<> storage_service::handle_state_removed(inet_address endpoint, std::vecto
 
 future<> storage_service::on_join(gms::inet_address endpoint, gms::endpoint_state_ptr ep_state, gms::permit_id pid) {
```

Comment:
I'm not sure how the existing code was supposed to work but handling of `RPC_READY` and `INTERNAL_IP` is done in conditional block, under `is_normal_token_owner(endpoint)`.
This change makes sure `STATUS` is handled before them so that an endpoint can become a normal token owner after handling `STATUS` by `on_change`, and after that `on_change` is called for `RPC_READY` and `INTERNAL_IP`.

I tried to explain that in the commit message, but apparently, unsuccessfully.
I'll try to clarify it.

Reply:
dropped from next version